s3/main.tf

```diff
@@ -85,6 +87,9 @@ resource "aws_s3_bucket" "bucket" {
       id                                     = "abort-incomplete-uploads"
       enabled                                = true
       abort_incomplete_multipart_upload_days = 7
+      expiration {
```

Comment:
This looks like it will make _all_ objects expire after 7 days. Am I misunderstanding the Terraform docs? What does this actually do?

Reply:
You're right, I wasn't looking at the bucket policy properly after I applied this. I was trying to stop terraform telling us there were changes to the bucket every time I ran it. I've found that if you set the expiration to 0 and set expired_object_delete_marker to false, terraform won't report changes to the bucket every time but the objects are not set to expire.